tools/firely-auth/firely-auth-settings.rst

```diff
@@ -0,0 +1,299 @@
+      "EC_Config": {
+          //"JWK_ES256": "<JSON Web Key>", // JSON Web Key of type EC with crv P-256
+          //"JWK_ES384": "<JSON Web Key>", // JSON Web Key of type EC with crv P-384
+          //"JWK_ES512": "<JSON Web Key>", // JSON Web Key of type EC with crv P-512
+          "SupportedAlgorithms": [
+              "ES256",
+              "ES384",
+              "ES512"
+          ]
+      }
+  }
+
+Firely Auth can work with multiple signature keys, used to sign access and other tokens. 
+
+- ``RSA_Config``: defines the RSA algorithms that are supported. In the config above all available algorithms are listed.
+  Inferno tests require at least RS256 for all Single Patient tests, and for Bulk Data Export a RS384 or higher is needed.
+
+  - ``RSA_JWK``: allows to provide a pre-generated JSON Web Key. If this is not provided, Firely Auth will generate a key.
+  - ``SupportedAlgorithms``: limit this list to the algorithms that you need in your setup. In the config above all available algorithms are listed.
+
+- ``EC_Config``: defines the EC (Elliptic Curve) algorithms that are supported. Inferno tests for Bulk Data Export require support for EC keys.
+
+  - ``JWK_ES*``: allows to provide a pre-generated JSON Web Key. If this is not provided, Firely Auth will generate a key for each of the supported algorithms.
+  - ``SupportedAlgorithms``: limit this list to the algorithms that you need in your setup. In the config above all available algorithms are listed.
+
+Note that a single RSA key can be used for all supported algorithms. However, an EC key is tied to a specific algorithm, therefore you can supply a key for each of the algorithms.
+
+For more background on JSON Web Keys see `RFC 7517 <see https://tools.ietf.org/html/rfc7517>`_.
+
+.. _firely_auth_settings_tokenintro:
+
+Token introspection
+^^^^^^^^^^^^^^^^^^^
+
+When using a :term:`reference token`, Firely Server must verify the token with Firely Auth.
+Not just any system can ask for inspection though, therefore we list the systems that can with a name and a secret.
+
+.. code-block:: json
+
+  "TokenIntrospectionConfig": {
+      "TokenIntrospectionResources": [{
+          "Name": "Firely Server",
+          "Secret": "<generate some hard to hack secret>"
+      }]
+  },
+
+This configuration is only needed if at least one :term:`client` is configured to use reference tokens, see :ref:`firely_auth_settings_tokentypes`.
+
+.. _firely_auth_settings_userstore:
+
+User store
+^^^^^^^^^^
+
+A :term:`user` must be able to authenticate to Firely Auth before granting permissions to a :term:`client`. 
+Therefore we register the users with Firely Auth. Firely Auth supports two types of stores: In memory and SQL Server.
+
+For the InMemory store, the users and their passwords are listed in plain text in this configuration. This is useful for testing, but not recommended for production use.
+
+The SqlServer store stores the users and their encrypted passwords in a MS SQL Server database. 
```

Comment:
It could be useful to mention here that the fhirUser claims can be configured here as well.

Reply:
That means that in line 221, under `InMemory - Claims` not only `patient` is valid, but also `fhirUser`. I would need to experiment with that again to document it correctly.

Reply:
Correct

Reply:
Processed change from `Claims` to `AdditionalClaims` and added `fhirUser`.
Added note to SqlServer store that `patient` and `fhirUser` claims can be added there as well.